lib/autoupdate/agent/telemetry.go

```diff
@@ -103,3 +113,59 @@ func hasParentDir(dir, parent string) (bool, error) {
 	}
 	return strings.HasPrefix(absDir, absParent), nil
 }
+
+// ReadHelloUpdaterInfo reads the updater config and generates a proto.UpdaterV2Info
+// that can be reported in the inventory hello message.
+// This function performs io operations, its usage must be cached
+// (the downstream inventory handler does this for us).
+func ReadHelloUpdaterInfo(ctx context.Context, log *slog.Logger, hostUUID string) (*proto.UpdaterV2Info, error) {
+	info := &proto.UpdaterV2Info{}
+
+	configPath := os.Getenv(updateConfigFileEnvVar)
+	if configPath == "" {
+		return nil, trace.Errorf("config file not specified")
+	}
+
+	cfg, err := readConfig(configPath)
+	if err != nil {
+		return nil, trace.Wrap(err, "reading config file %s", configPath)
+	}
+
+	info.UpdateGroup = cfg.Spec.Group
+	if p := cfg.Status.IDFile; p != "" {
+		machineID, err := os.ReadFile(systemdMachineIDFile)
+		if err != nil && !errors.Is(err, fs.ErrNotExist) {
+			log.WarnContext(ctx, "Failed to read systemd machine ID.", "path", systemdMachineIDFile, errorKey, err)
+			log.WarnContext(ctx, "Updater ID may be inaccurate for tracking.")
+			machineID = nil
+		}
+		id, err := findDBPIDUUID(p, []byte(hostUUID), machineID, true)
```

Comment:
Is this guaranteed to return a UUID? Teleport host IDs can also be EC2 instance IDs, unfortunately ._.

Reply:
I think that we take any string as the host UUID, convert it to a byte slice, and put it into the ASN.1 thing that gets hashed. So at the end we always have a UUID, and we don't care about the host ID being strictly a UUID.

Reply:
DBPIDs are guaranteed to be 16-byte deterministic UUIDv5s.